### PR TITLE
doc: make implicity the users must set envtest

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
Update the title to make clear to the users that the must configure the envtest

**Motivation for the change:**
The issue raised: #3692

### CODE REVIEW

**Comment by camilamacedo86** [comment]
Hi @jmrodri and @jmccormick2001, 

WDYT about this change in the doc until we are able to solve it when v3+ plugin be supported?

File: website/content/en/docs/building-operators/golang/quickstart.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Create a simple Memcached API:
 operator-sdk create api --group cache --version v1 --kind Memcached --resource=true --controller=true
 ```
 
-### Configuring your test environment
+### You must configure your test environment
```
Comment (jmccormick2001) [inline website/content/en/docs/building-operators/golang/quickstart.md:37]: Configure Test Environment (REQUIRED for building the operator)

Comment (camilamacedo86) [inline website/content/en/docs/building-operators/golang/quickstart.md:37]:
```suggestion
### Configure Test Environment (REQUIRED for building the operator)
```

Comment (joelanford) [reply website/content/en/docs/building-operators/golang/quickstart.md]: I disagree with changing the headers. Can we add a sentence in this section (and possibly in the section that includes `make docker-build`) about the reasons for the necessity of this step.

File: website/content/en/docs/building-operators/golang/quickstart.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ Create a simple Memcached API:
 operator-sdk create api --group cache --version v1 --kind Memcached --resource=true --controller=true
 ```
 
-### Configuring your test environment
+### Configure Test Environment (REQUIRED for building the operator)
```
Comment (camilamacedo86) [inline website/content/en/docs/building-operators/golang/quickstart.md:37]:
```suggestion
### Configuring your test environment 
**(REQUIRED for building the operator)**
```

File: website/content/en/docs/building-operators/golang/quickstart.md (hunk 1 above)
Comment (camilamacedo86) [reply website/content/en/docs/building-operators/golang/quickstart.md]: let us know wdyt now @joelanford ?

**Comment by jmccormick2001** [comment]
one thing you might consider, instead of having a 'setup envtest' section, you might instead just reference how to install the dependencies required to do the docker-build, I found that yesterday in testing some things, you can just follow https://book-v1.book.kubebuilder.io/getting_started/installation_and_setup.html to install the kubebuilder dependency, then proceed.

**Comment by camilamacedo86** [comment]
Hi @jmccormick2001 
 
> one thing you might consider, instead of having a 'setup envtest' section, you might instead just reference how to install the dependencies required to do the docker-build, I found that yesterday in testing some things, you can just follow https://book-v1.book.kubebuilder.io/getting_started/installation_and_setup.html to install the kubebuilder dependency, then proceed.

It is true, teh setup is not required If users install the kubebuilder. However, kubebuilder should NOT be a pre-requirement for users are able to use SDK. The setup in the quick-start will NOT install kubebuilder just the required binaries in the project instead of also requires sudo permission as described in the Kubebuilder installation doc. 

With the implementation in the PR https://github.com/kubernetes-sigs/kubebuilder/pull/1626 we are doing this update in the Makefile that is generated by the GO plugin which also is required for we achieve other requirements for Kubebuilder as well. See that in the long term we should no longer provide these binaries with kubebuilder https://github.com/kubernetes-sigs/kubebuilder/issues/686.  Anyway, the step in the doc will be required for a while since the change is valid just for v3+ plugin and we are using in SDK the v2 plugin version. 

Is it make sense?

**Comment by camilamacedo86** [comment]
Close this one in favor of: https://github.com/operator-framework/operator-sdk/pull/3983

**Pull request closed without merging**